pyccel/ast/c_concepts.py

```diff
@@ -55,38 +55,50 @@ class CStringExpression(Basic):
         )
     """
     __slots__  = ('_expression',)
-    _attribute_nodes  = ()
+    _attribute_nodes  = ('_expression',)
 
     def __init__(self, *args):
-        super().__init__()
         self._expression = []
+        super().__init__()
         for arg in args:
-            self += arg
+            self.append(arg)
 
```

Comment:
Wouldn't this do the same thing if it was written as:
```
def __init__(self, *args):
    self._expression = args
    super().__init__()
```
?

Reply:
Using the first one makes sure that you can only add LiteralString, CMacro or CStringExpression as args